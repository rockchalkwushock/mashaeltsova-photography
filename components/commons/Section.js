import { styles } from '../../lib'

const Section = props => (
  <section
    className={
      ('header' in props && 'header') ||
      ('navigation' in props && 'navigation') ||
      ('about' in props && 'about') ||
      ('services' in props && 'services') ||
      ('gallery' in props && 'gallery') ||
      ('booking' in props && 'booking') ||
      ('footer' in props && 'footer')
    }
  >
    {props.children}
    <style jsx>{`
      section {
        background-color: ${styles.colors.powder};
      }
      .header,
      .about,
      .services,
      .gallery,
      .booking {
        grid-column: span 7;
      }
      .header {
        height: 500px;
      }
      .about,
      .gallery {
        height: 250px;
      }
      .services {
        height: 150px;
      }
      .booking {
        height: 300px;
      }
      @media (min-width: 768px) {
        .about {
          grid-column: span 4;
        }
        .services {
          grid-column: span 3;
          height: 250px;
        }
        .gallery,
        .booking {
          height: 500px;
        }
      }
      @media (min-width: 1025px) {
        .about {
          grid-column: span 7;
        }
        .services {
          grid-column: 5 / 8;
          height: 500px;
        }
        .gallery {
          grid-column: span 7;
          grid-row: 4 / 5;
        }
        .booking {
          grid-column: 1 / 5;
          grid-row: 5 / 6;
        }
      }
    `}</style>
  </section>
)

export default Section
