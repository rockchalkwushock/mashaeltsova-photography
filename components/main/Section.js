import PropTypes from 'prop-types'

import { styles } from '../../lib'

const Section = ({ children, className }) => (
  <section className={className}>
    {children}
    <style jsx>{`
      .about {
        background-color: ${styles.colors.lemon};
        box-shadow: 0 0 20px rgba(54, 69, 79, 0.7);
        color: ${styles.colors.charcoal};
        grid-area: about;
        grid-column: 1 / 5;
        padding: 0.75em;
        text-align: left;
      }
      .services {
        background-color: ${styles.colors.lemon};
        box-shadow: 0 0 20px rgba(54, 69, 79, 0.7);
        color: ${styles.colors.charcoal};
        display: flex;
        flex-direction: column;
        grid-area: services;
        grid-column: 1 / 5;
        padding: 0.75em;
        text-align: left;
      }
      .gallery {
        background-color: ${styles.colors.peach};
        box-shadow: 0 0 20px rgba(54, 69, 79, 0.7);
        display: flex;
        grid-area: gallery;
        grid-column: 1 / 5;
        height: 250px;
        justify-content: center;
      }
      .booking {
        box-shadow: 0 0 20px rgba(54, 69, 79, 0.7);
        grid-area: booking;
        grid-column: 1 / 5;
      }
      @media (min-width: 540px) {
        .gallery {
          height: 360px;
        }
      }
      @media (min-width: 768px) {
        .about {
          grid-column: 1 / 3;
          grid-row: 6 / 8;
        }
        .services {
          grid-column: 3 / 5;
          grid-row: 6 / 8;
        }
        .gallery {
          grid-column: 1 / 5;
          grid-row: 1 / 6;
        }
        .booking {
          display: grid;
          grid-template-columns: 1fr 2fr;
          grid-column: 1 / 5;
          grid-row: 8 / 12;
        }
      }
      @media (min-width: 1024px) {
        .gallery {
          height: 425px;
        }
      }
    `}</style>
  </section>
)

Section.propTypes = {
  className: PropTypes.string.isRequired
}

export default Section
