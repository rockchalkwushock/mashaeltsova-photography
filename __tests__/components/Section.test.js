import React from 'react'
import { render } from 'enzyme'

import Section from '../../components/main/Section'

describe('Component: Section', () => {
  test('Renders without exploding as .about', () => {
    const tree = render(<Section className="about" />)
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding as .services', () => {
    const tree = render(<Section className="services" />)
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding as .gallery', () => {
    const tree = render(<Section className="gallery" />)
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding as .booking', () => {
    const tree = render(<Section className="booking" />)
    expect(tree).toMatchSnapshot()
  })
})
