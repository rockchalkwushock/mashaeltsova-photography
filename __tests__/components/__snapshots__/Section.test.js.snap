// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Section Renders without exploding as .about 1`] = `
<section
  class="jsx-2104976944 about"
/>
`;

exports[`Component: Section Renders without exploding as .booking 1`] = `
<section
  class="jsx-2104976944 booking"
/>
`;

exports[`Component: Section Renders without exploding as .gallery 1`] = `
<section
  class="jsx-2104976944 gallery"
/>
`;

exports[`Component: Section Renders without exploding as .services 1`] = `
<section
  class="jsx-2104976944 services"
/>
`;
